.github/workflows/main.yml

```diff
@@ -19,14 +19,16 @@ jobs:
         wget $(curl -s 'https://ziglang.org/download/index.json' | python3 -c "import sys, json; print(json.load(sys.stdin)['master']['x86_64-linux']['tarball'])")
         sudo apt-get install mtools
         tar -xvf zig*
-    - name: Build kernel
-      run: zig*/zig build ${{ matrix.build_mode }}
-    - name: Run unit tests
-      run: zig*/zig build test ${{ matrix.build_mode }}
     - name: Install qemu
       run: |
         sudo apt-get update
         sudo apt-get install qemu qemu-system --fix-missing
+    - name: chmod
+      run: chmod +x ./makeiso.sh
+    - name: Build kernel
```

Comment:
I had the build and test steps before downloading qemu as they don't depend on it and it means that if installing qemu fails for whatever reason, we can still test and build. I can see an advantage of installing qemu first though as it means installation failures will appear sooner and mean you can re-run sooner. What do you think?

Reply:
The testing is now done in qemu, so need to download before testing. Also thought that the download steps could be grouped together

Reply:
Yep that makes sense.